flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesConfigMapSharedInformer.java

```diff
@@ -21,18 +21,17 @@
 import org.apache.flink.kubernetes.kubeclient.KubernetesConfigMapSharedWatcher;
 
 import io.fabric8.kubernetes.api.model.ConfigMap;
-import io.fabric8.kubernetes.api.model.ConfigMapList;
 import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
 
 import java.util.Map;
 
 /** The shared informer for {@link ConfigMap}, it can be used as a shared watcher. */
 public class KubernetesConfigMapSharedInformer
-        extends KubernetesSharedInformer<ConfigMap, ConfigMapList, KubernetesConfigMap>
+        extends KubernetesSharedInformer<ConfigMap, KubernetesConfigMap>
         implements KubernetesConfigMapSharedWatcher {
 
     public KubernetesConfigMapSharedInformer(
             NamespacedKubernetesClient client, Map<String, String> labels) {
-        super(client, ConfigMap.class, ConfigMapList.class, labels, KubernetesConfigMap::new);
+        super(client, client.configMaps().withLabels(labels), KubernetesConfigMap::new);
```

Comment:
Should we add null check here like before in the `KubernetesSharedInformer`?

```
Preconditions.checkArgument(
                !CollectionUtil.isNullOrEmpty(labels), "Labels must not be null or empty");
```

Reply:
Make sense. I will add such sanity check here.